packages/injectable/ogre-tools-injectable.d.ts

```diff
@@ -1,160 +1,144 @@
-        InferFromInjectable<TInjectable>[1]
-      >
-    ): TReturnAsPromise extends true
-      ? Promise<InferFromInjectable<TInjectable>[0]>
-      : InferFromInjectable<TInjectable>[0];
-
-    inject<TInjectionToken extends InjectionToken<unknown, unknown>>(
-      injectionToken: TInjectionToken,
-      ...instantiationParameter: TentativeTuple<
-        TInjectionToken['instantiationParameter']
-      >
-    ): TReturnAsPromise extends true
-      ? Promise<TInjectionToken['template']>
-      : TInjectionToken['template'];
-
-    injectMany<TInjectionToken extends InjectionToken<unknown, unknown>>(
-      injectionToken: TInjectionToken,
-      ...instantiationParameter: TentativeTuple<
-        TInjectionToken['instantiationParameter']
-      >
-    ): TReturnAsPromise extends true
-      ? Promise<TInjectionToken['template'][]>
-      : TInjectionToken['template'][];
+    inject: Inject<TReturnAsPromise>;
+
+    injectMany<Instance, InstantiationParam = void>(
+      injectionToken: InjectionToken<Instance, InstantiationParam>,
+      instantiationParameter: InstantiationParam[],
+    ): AsyncReturnable<TReturnAsPromise, Instance>;
   }
 
-  type InferFromToken<T> = T extends InjectionToken<
-    infer TInstance,
-    infer TInstantiationParameter
-  >
-    ? [TInstance, TInstantiationParameter]
-    : never;
+  export interface ILifecycle<InstantiationParam = void, Key extends string | number | symbol = string | number | symbol> {
+    getInstanceKey: (di: DiContainer, params: InstantiationParam) => Key;
+  }
 
-  export interface ILifecycle {
-    getInstanceKey: (di: DiContainer, param: unknown) => string | number;
+  export interface KeyedSingletonOptions<InstantiationParam = void> {
+    getInstanceKey: (di: DiContainer, params: InstantiationParam) => string | number | symbol;
   }
 
+  export type KeyedSingleton = <InstantiationParam = void>(keyedSingletonOptions: KeyedSingletonOptions<InstantiationParam>) => ILifecycle<InstantiationParam>;
+
+  const singleton: unique symbol;
+  const transient: unique symbol;
+
   export const lifecycleEnum: {
-    singleton: ILifecycle;
+    singleton: ILifecycle<[], typeof singleton>;
 
-    keyedSingleton: <TInstantiationParameter>(keyedSingletonOptions: {
-      getInstanceKey: (
-        di: DiContainer,
-        instantiationParameter: TInstantiationParameter,
-      ) => string | number;
-    }) => ILifecycle;
+    keyedSingleton: KeyedSingleton;
 
-    transient: ILifecycle;
+    transient: ILifecycle<[], typeof transient>;
   };
 
   export function createContainer(...getRequireContexts: any[]): DiContainer;
 
-  export function registerErrorMonitoring(di: DiContainer);
+  export function registerErrorMonitoring(di: DiContainer): void;
 
   export const errorMonitorInjectionToken: InjectionToken<
     (error: {
       context: { id: string; instantiationParameter: any }[];
       error: any;
     }) => void | Promise<void>,
-    void
+    []
```

Comment:
Empty array for no `instantiationParameter` is odd, as there is no possibility for more than one parameter. Therefore `void` is more fitting way to convey this information. 

Hmm. No parameter when irrelevant would be even better, I think.

Reply:
This was a mistake, should stay as `void` or even just use the default of `void.